spec/Section 2 -- Source Schema.md

```diff
@@ -1,244 +1,399 @@
+
 ## Directives
 
-### @entityResolver
+### @lookup
 
 ```graphql
-directive @entityResolver on FIELD_DEFINITION
+directive @lookup on FIELD_DEFINITION
 ```
 
-Entity resolvers are fields on the query root type of a subgraph that can
-resolve an entity by a stable key. The stable key is defined by the arguments of
-the field.
+The `@lookup` directive is used within a _source schema_ to specify output
+fields that can be used by the _distributed GraphQL executor_ to resolve an
+entity by a stable key.
+
+The stable key is defined by the arguments of the field. Each argument must
+match a field on the return type of the lookup field.
+
+Source schemas can provide multiple lookup fields for the same entity that
+resolve the entity by different keys.
+
+In this example, the source schema specifies that the `Product` entity can be
+resolved with the `productById` field or the `productByName` field. Both lookup
+fields are able to resolve the `Product` entity but do so with different keys.
 
 ```graphql example
-extend type Query {
-  version: Int # NOT an entity resolver.
-  personById(id: ID!): Person @entityResolver
+type Query {
+  version: Int # NOT a lookup field.
+  productById(id: ID!): Product @lookup
+  productByName(name: String!): Product @lookup
 }
 
-extend type Person {
-  id: ID! # matches the argument of personById
+type Product @key(fields: "id") @key(fields: "name") {
+  id: ID!
+  name: String!
 }
 ```
 
-The arguments of an entity resolver field must match fields of the returning
-type.
+The arguments of a lookup field must correspond to fields specified as an entity
+key with the `@key` directive on the entity type.
 
 ```graphql example
-extend type Query {
-  node(id: ID!): Node @entityResolver
+type Query {
+  node(id: ID!): Node @lookup
 }
 
-interface Node {
+interface Node @key(fields: "id") {
   id: ID!
 }
 ```
 
-When an entity resolver returns an interface all implementing types are inferred
-as entities.
+Lookup fields may return object, interface, or union types. In case a lookup
+field returns an abstract type (interface type or union type), all possible
+object types are considered entities and must have keys that correspond with the
+fields argument signature.
 
 ```graphql example
-extend type Query {
-  entityById(id: ID!, categoryId: Int): Entity @entityResolver
+type Query {
+  product(id: ID!, categoryId: Int): Product @lookup
+}
+
+union Product = Electronics | Clothing
+
+type Electronics @key(fields: "id categoryId") {
+  id: ID!
+  categoryId: Int
+  name: String
+  brand: String
+  price: Float
+}
+
+type Clothing @key(fields: "id categoryId") {
+  id: ID!
+  categoryId: Int
+  name: String
+  size: String
+  price: Float
+}
+```
+
+The following example shows an invalid lookup field as the `Clothing` type does
+not declare a key that corresponds with the lookup field's argument signature.
+
+```graphql counter-example
+type Query {
+  product(id: ID!, categoryId: Int): Product @lookup
```

Comment:
as discussed during last meeting I believe this is actually valid?

Reply:
Yes ... I am gonna rewrite this part.